### ISSUE TITLE
DownloadDirectory may not work on custom systems due to internal FileListing

### ISSUE THREAD

**Issue by Adhara3** [robinrodricks/FluentFTP#772]
**FTP OS:** Embedded

**FTP Server:** Custom

**Computer OS:** Windows

**FluentFTP Version:** 35.0.5

DownloadDirectory is internally performing a `this.GetListing(remoteFolder, FtpListOption.Size | FtpListOption.Recursive)` so using specific parameters that may not work as expected on all system. 
Example: `FtpListOption.NoPath` may be needed in order to work.

This should be probably part of the FtpServerBase class. 
Please note that the enum provides some behavioral flags (Recursive) while others are more archictural (NoPath, ForceList) and the BaseServer class should fill the latter kind

Regards
A

**Comment by robinrodricks** [comment]
Good idea, any suggestions how to implement this? A new property/method in FtpServerBase?

**Comment by Adhara3** [comment]
Hi,

yes I would probably adda new method with a signature like the following
`FtpListOption DecorateFileListOptions(FtpListOption options)`

I can do the pull request if you want

Cheers
A

**Comment by robinrodricks** [comment]
@Adhara3  please don't need to ask my permission to make pull requests, its hard to check so many active tickets. If you have any suggestion, please just file a PR and we can take it forward. I accept most PRs straightaway unless they cause a major conflict.
